test/libsyncengine/update_detection/update_detector/testupdatetreeworker.cpp

```diff
@@ -278,6 +278,80 @@ void TestUpdateTreeWorker::testUtilsFunctions() {
     CPPUNIT_ASSERT(_updateTree->getNodeByPath(newPath)->id() == "id4111");
 }
 
+void TestUpdateTreeWorker::testUpdateTmpFileNode() {
+    auto deleteOp =
+        std::make_shared<FSOperation>(OperationTypeDelete, "id51", NodeTypeFile, 1654788552, 1654788552, 12345, "Dir 5/File 5.1");
+    auto createOp = std::make_shared<FSOperation>(OperationTypeCreate, "id511", NodeTypeFile, 1654798336, 1654798336, 12345,
+                                                  "Dir 5/File 5.1");
+
+    {
+        auto newNode = _updateTreeWorker->getOrCreateNodeFromExistingPath("Dir 5/File 5.1");
+        CPPUNIT_ASSERT(newNode->id()->substr(0, 4) == "tmp_");
+        CPPUNIT_ASSERT(newNode->isTmp());
+
+        _updateTreeWorker->updateTmpFileNode(newNode, createOp, deleteOp, OperationTypeEdit);
+        CPPUNIT_ASSERT_EQUAL(NodeId("id511"), *newNode->id());
+        CPPUNIT_ASSERT_EQUAL(*newNode->createdAt(), createOp->createdAt());
+        CPPUNIT_ASSERT_EQUAL(*newNode->lastmodified(), createOp->lastModified());
+        CPPUNIT_ASSERT_EQUAL(newNode->size(), createOp->size());
+        CPPUNIT_ASSERT(newNode->hasChangeEvent(OperationTypeEdit));
+        CPPUNIT_ASSERT_EQUAL(deleteOp->nodeId(), *newNode->previousId());
+        CPPUNIT_ASSERT(!newNode->isTmp());
+        CPPUNIT_ASSERT(_updateTree->nodes()[createOp->nodeId()] == newNode);
+    }
+
+    _updateTree->deleteNode(NodeId("id511"));
+
+    {
+        auto newNode = _updateTreeWorker->getOrCreateNodeFromExistingPath("Dir 5/File 5.1");
+        CPPUNIT_ASSERT(newNode->id()->substr(0, 4) == "tmp_");
+        CPPUNIT_ASSERT(newNode->isTmp());
+
+        _updateTreeWorker->updateTmpFileNode(newNode, deleteOp, deleteOp, OperationTypeDelete);
+        CPPUNIT_ASSERT_EQUAL(NodeId("id51"), *newNode->id());
+        CPPUNIT_ASSERT_EQUAL(*newNode->createdAt(), deleteOp->createdAt());
+        CPPUNIT_ASSERT_EQUAL(*newNode->lastmodified(), deleteOp->lastModified());
+        CPPUNIT_ASSERT_EQUAL(newNode->size(), createOp->size());
+        CPPUNIT_ASSERT(newNode->hasChangeEvent(OperationTypeDelete));
+        CPPUNIT_ASSERT(!newNode->isTmp());
+        CPPUNIT_ASSERT(_updateTree->nodes()[deleteOp->nodeId()] == newNode);
+    }
+}
+
+void TestUpdateTreeWorker::testHandleCreateOperationsWithSamePath() {
+    setUpUpdateTree();
+
+    // Regular case: success
+    _operationSet->insertOp(std::make_shared<FSOperation>(OperationTypeDelete, "id4111", NodeTypeFile, 1654798667, 1654798667,
+                                                          12345, "Dir 4/Dir 4.1/Dir 4.1.1/File 4.1.1.1"));
+    _operationSet->insertOp(std::make_shared<FSOperation>(OperationTypeDelete, "id51", NodeTypeFile, 1654788552, 1654788552,
+                                                          12345, "Dir 5/File 5.1"));
+    _operationSet->insertOp(std::make_shared<FSOperation>(OperationTypeCreate, "id511", NodeTypeFile, 1654798336, 1654798336,
+                                                          12345, "Dir 5/File 5.1"));
```

Comment:
I think everywhere else in this file, 3 digit in the code means 3 level deep in the tree (a file with ID 'id511' should have a path such as "Dir 5/Dir 5.1/File 5.1.1"). Can we imagine a different ID in order to make it easier to read?

Reply:
You are right, this is not consistent. It originates from a copy-paste, so there are other instances of this mismatch in the same file. I have made a proposal in the next commit to fix all such issues.